### PR TITLE
qsd_block_stream:do stream under qsd

### PR DESCRIPTION
**Pull request by aliang123**
Do stream under qsd

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2133675

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (90.18 s)

**Comment by aliang123** [comment]
@zhencliu @XueqiangWei, please help to review, thanks.

**Comment by aliang123** [comment]
depends on https://github.com/autotest/tp-qemu/pull/3497

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (92.72 s)

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (86.29 s)

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (86.91 s)

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (87.17 s)

File: qemu/tests/cfg/qsd_block_stream.cfg [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    start_vm = no
+    storage_pools = default
+    storage_type_default = "directory"
+    storage_pool = default
+    base_tag = "stg1"
+    snapshot_tag = sn1
+    image_name_sn1 = sn1
+    image_format_sn1 = qcow2
+    device = "fmt_stg1"
+    format = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/qsd_block_stream.cfg:14]: we can remove these two lines

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (86.97 s)

File: qemu/tests/cfg/qsd_block_stream.cfg (hunk 1 above)
Comment (aliang123) [reply qemu/tests/cfg/qsd_block_stream.cfg]: @zhencliu, have updated, thanks.

**Review by zhencliu — APPROVED** [review]

**Comment by zhencliu** [comment]
hi @aliang123 , for these qsd block job patches, I found dup codes, the only difference is the base class uses vm while the test case itself uses qsd, a good way may enhance the base block job classes to support not only vm but also qsd, but currently, IMO, as we will not introduce more block job cases for qsd, we don't need to do the change. I just give a reminder here, that we have to consider this if in future we need to add more qsd block job cases. Also cc @XueqiangWei

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.qsd_block_stream.q35: PASS (135.02 s)

**Comment by aliang123** [comment]
> hi @aliang123 , for these qsd block job patches, I found dup codes, the only difference is the base class uses vm while the test case itself uses qsd, a good way may enhance the base block job classes to support not only vm but also qsd, but currently, IMO, as we will not introduce more block job cases for qsd, we don't need to do the change. I just give a reminder here, that we have to consider this if in future we need to add more qsd block job cases. Also cc @XueqiangWei

Yes, zhenchao, if qsd is integrited with libvirt, we need to test all block job cases on both qsd and qemu, then we will optimize the design and make it compatible with both vm.qmp.monitor and qsd.monitor, thanks.

**Pull request merged**